src/communication/vchannel.cpp

```diff
@@ -53,18 +53,24 @@ aos::Error VChannel::Close()
 
 aos::Error VChannel::Read(aos::Array<uint8_t>& data, size_t size)
 {
-    auto err = data.Resize(size);
-    if (!err.IsNone()) {
-        return AOS_ERROR_WRAP(err);
-    }
+    size_t read = 0;
 
-    auto ret = vch_read(&mReadHandle, data.Get(), size);
-    if (ret < 0) {
-        return AOS_ERROR_WRAP(ret);
+    while (read < size) {
+        auto err = data.Resize(size);
+        if (!err.IsNone()) {
+            return AOS_ERROR_WRAP(err);
+        }
+
+        auto ret = vch_read(&mReadHandle, data.Get() + read, size - read);
+        if (ret < 0) {
+            return AOS_ERROR_WRAP(ret);
+        }
+
+        read += ret;
     }
 
-    if (static_cast<size_t>(ret) != size) {
-        err = data.Resize(ret);
+    if (read != size) {
```

Comment:
not sure this check is required anymore.

Reply:
I would like to return actual amount of data.